### PR TITLE
Develop nutty modify framework

### PR DESCRIPTION
**Pull request by nutty898**
Adjust framework.

### CODE REVIEW

File: cocos/editor-support/cocostudio/CCSGUIReader.cpp [hunk 1]
```diff
@@ -218,78 +218,78 @@ UIWidget* WidgetPropertiesReader0250::createWidget(JsonDictionary* data, const c
     return widget;
 }
 
-UIWidget* WidgetPropertiesReader0250::widgetFromJsonDictionary(JsonDictionary *data)
+Widget* WidgetPropertiesReader0250::widgetFromJsonDictionary(JsonDictionary *data)
```
Comment (dumganhar) [inline cocos/editor-support/cocostudio/CCSGUIReader.cpp:221]: What does `0250` mean?

File: cocos/gui/UIListView.cpp [hunk 2]
```diff
@@ -362,34 +359,38 @@ void UIListView::setDirection(SCROLLVIEW_DIR dir)
...
-    {
-        return;
-    }
-    cocos2d::ccArray* arrayItems = _items->data;
-    int length = arrayItems->num;
+    int length = _items.size();
     for (int i=0; i<length; i++)
     {
-        UIWidget* item = (UIWidget*)(arrayItems->arr[i]);
+        Widget* item = (Widget*)(_items.at(i));
```
Comment (dumganhar) [inline cocos/gui/UIListView.cpp]: The `cast` is not needed.

File: cocos/gui/UIPageView.cpp [hunk 3]
```diff
@@ -163,40 +161,39 @@ void UIPageView::insertPage(UILayout* page, int idx)
...
             page->setSize(pvSize);
         }
-        cocos2d::ccArray* arrayPages = _pages->data;
-        int length = arrayPages->num;
+        int length = _pages.size();
         for (int i=(idx+1); i<length; i++) {
-            UIWidget* behindPage = dynamic_cast<UIWidget*>(arrayPages->arr[i]);
-            cocos2d::Point formerPos = behindPage->getPosition();
-            behindPage->setPosition(cocos2d::Point(formerPos.x+getSize().width, 0));
+            Widget* behindPage = dynamic_cast<Widget*>(_pages.at(i));
```
Comment (dumganhar) [inline cocos/gui/UIPageView.cpp]: `_pages` is defined as `Vector<Widget*>`, So the `cast` is not needed.

File: cocos/gui/UIPageView.cpp [hunk 4]
```diff
@@ -207,86 +204,87 @@ void UIPageView::removePage(UILayout* page)
...
+    if (_pages.size() <= 0)
     {
         _leftChild = nullptr;
         _rightChild = nullptr;
         return;
     }
-    _leftChild = dynamic_cast<UIWidget*>(_pages->getObjectAtIndex(0));
-    _rightChild = dynamic_cast<UIWidget*>(_pages->getLastObject());
+    _leftChild = dynamic_cast<Widget*>(_pages.at(0));
+    _rightChild = dynamic_cast<Widget*>(_pages.at(_pages.size()-1));
```
Comment (dumganhar) [inline cocos/gui/UIPageView.cpp]: `_pages` is defined as `Vector<Widget*>`, So the `cast` is not needed.

File: cocos/gui/UIPageView.cpp [hunk 5]
```diff
@@ -378,52 +375,60 @@ void UIPageView::update(float dt)
...
-void UIPageView::movePages(float offset)
+void PageView::movePages(float offset)
 {
-    cocos2d::ccArray* arrayPages = _pages->data;
-    int length = arrayPages->num;
+    int length = _pages.size();
     for (int i = 0; i < length; i++)
     {
-        UIWidget* child = (UIWidget*)(arrayPages->arr[i]);
+        Widget* child = (Widget*)(_pages.at(i));
```
Comment (dumganhar) [inline cocos/gui/UIPageView.cpp]: `_pages` is defined as `Vector<Widget*>`, So the `cast` is not needed.

File: cocos/gui/UIPageView.cpp [hunk 6]
```diff
@@ -492,17 +492,17 @@ void UIPageView::handleMoveLogic(const cocos2d::Point &touchPoint)
...
-void UIPageView::handleReleaseLogic(const cocos2d::Point &touchPoint)
+void PageView::handleReleaseLogic(const Point &touchPoint)
 {
-    if (_pages->count() <= 0)
+    if (_pages.size() <= 0)
     {
         return;
     }
-    UIWidget* curPage = dynamic_cast<UIWidget*>(_pages->getObjectAtIndex(_curPageIdx));
+    Widget* curPage = dynamic_cast<Widget*>(_pages.at(_curPageIdx));
```
Comment (dumganhar) [inline cocos/gui/UIPageView.cpp]: `_pages` is defined as `Vector<Widget*>`, So the `cast` is not needed.

File: cocos/gui/UIWidget.cpp [hunk 7]
```diff
@@ -102,226 +84,187 @@ UIWidget* UIWidget::create()
...
-    _renderer->retain();
-    _renderer->setZOrder(_widgetZOrder);
-    cocos2d::RGBAProtocol* renderRGBA = DYNAMIC_CAST_CCRGBAPROTOCOL;
-    if (renderRGBA)
+    if (NodeRGBA::init())
     {
-        renderRGBA->setCascadeColorEnabled(true);
-        renderRGBA->setCascadeOpacityEnabled(true);
+        _layoutParameterDictionary = Dictionary::create();
+        CC_SAFE_RETAIN(_layoutParameterDictionary);
```
Comment (dumganhar) [inline cocos/gui/UIWidget.cpp]: Don't use `Dictionary` anymore, please use `std::unordered_map` or `cocos2d::Map<K, V>` instead.

File: cocos/gui/UIWidget.cpp [hunk 8]
```diff
@@ -487,72 +490,76 @@ void UIWidget::ignoreContentAdaptWithSize(bool ignore)
...
+        listener->setSwallowTouches(true);
+        listener->onTouchBegan = CC_CALLBACK_2(Widget::onTouchBegan, this);
+        listener->onTouchMoved = CC_CALLBACK_2(Widget::onTouchMoved, this);
+        listener->onTouchEnded = CC_CALLBACK_2(Widget::onTouchEnded, this);
+        listener->onTouchCancelled = CC_CALLBACK_2(Widget::onTouchCancelled, this);
+        _eventDispatcher->addEventListenerWithSceneGraphPriority(listener, this);
+    }
+    else
+    {
+        _eventDispatcher->removeEventListeners(EventListener::Type::TOUCH_ONE_BY_ONE);
```
Comment (dumganhar) [inline cocos/gui/UIWidget.cpp]: ERROR: `removeEventListeners(EventListener::Type::TOUCH_ONE_BY_ONE);` will remove all listeners with `ONE_BY_ONE` type.
You should save the listener as `Widget`'s member, check whether is valid.
Invoking `EventDispatcher::removeEventListener(_touchlistener);` instead.

**Comment by dumganhar** [comment]
Please update Android.mk and CMakeLists.txt.
Android and linux build failed. https://travis-ci.org/cocos2d/cocos2d-x/builds/15645094

File: cocos/2d/CCNode.cpp [hunk 9]
```diff
@@ -1442,6 +1442,20 @@ bool NodeRGBA::init()
     return false;
 }
 
+NodeRGBA * NodeRGBA::create(void)
+{
+	NodeRGBA * pRet = new NodeRGBA();
```
Comment (minggo) [inline cocos/2d/CCNode.cpp:1447]: Don't use hungarian coding style.

File: samples/Cpp/TestCpp/Classes/ExtensionsTest/CocoStudioGUITest/UITextFieldTest/UITextFieldTest.h [hunk 10]
```diff
@@ -63,6 +63,6 @@ class UITextFieldTest_Password : public UIScene
     
 protected:
     UI_SCENE_CREATE_FUNC(UITextFieldTest_Password)
-    UILabel* m_pDisplayValueLabel;
+    cocos2d::gui::Label* m_pDisplayValueLabel;
```
Comment (minggo) [inline samples/Cpp/TestCpp/Classes/ExtensionsTest/CocoStudioGUITest/UITextFieldTest/UITextFieldTest.h:66]: Don't use hungarian coding style.

File: cocos/gui/UISlider.h [hunk 11]
```diff
@@ -161,25 +164,18 @@ class UISlider : public UIWidget
...
-    //override "onTouchMoved" method of widget.
-    virtual void onTouchMoved(const cocos2d::Point &touchPoint) override;
-    
-    //override "onTouchEnded" method of widget.
-    virtual void onTouchEnded(const cocos2d::Point &touchPoint) override;
+    void addEventListenerSlider(Object* target,SEL_SlidPercentChangedEvent selector);
     
-    //override "onTouchCancelled" method of widget.
-    virtual void onTouchCancelled(const cocos2d::Point &touchPoint) override;
+    virtual bool onTouchBegan(Touch *touch, Event *unused_event) override;
```
Comment (minggo) [inline cocos/gui/UISlider.h:169]: `unused_event` -> `unusedEvent`

File: cocos/gui/UIWidget.h [hunk 12]
```diff
@@ -944,68 +719,52 @@ class UIWidget : public cocos2d::Object
...
-    int _widgetZOrder;        ///< z-order value that affects the draw order and touch order
-    cocos2d::Point _anchorPoint;      ///< anchor point normalized
-    UIWidget* _widgetParent;  ///< parent of widget
     BrightStyle _brightStyle; ///< bright style
     bool _updateEnabled;      ///< is "update" method scheduled
-    cocos2d::Node* _renderer;        ///< base renderer
-    cocos2d::Point _touchStartPos;    ///< touch began point
-    cocos2d::Point _touchMovePos;     ///< touch moved point
-    cocos2d::Point _touchEndPos;      ///< touch ended point
+    NodeRGBA* _renderer;        ///< base renderer
```
Comment (minggo) [inline cocos/gui/UIWidget.h:697]: `NodeRGBA` is deprecated, i think you don't need it now. All interfaces are merged into `Node`.

File: cocos/editor-support/cocostudio/CCSGUIReader.cpp (hunk 1 above)
Comment (nutty898) [inline cocos/editor-support/cocostudio/CCSGUIReader.cpp:221]: That means it is a reader for the framework whose version is before 0.2.5.0

File: cocos/2d/CCNode.cpp (hunk 9 above)
Comment (nutty898) [inline cocos/2d/CCNode.cpp:1447]: ok

File: samples/Cpp/TestCpp/Classes/ExtensionsTest/CocoStudioGUITest/UITextFieldTest/UITextFieldTest.h (hunk 10 above)
Comment (nutty898) [inline samples/Cpp/TestCpp/Classes/ExtensionsTest/CocoStudioGUITest/UITextFieldTest/UITextFieldTest.h:66]: ok

File: cocos/gui/UIWidget.h (hunk 12 above)
Comment (nutty898) [inline cocos/gui/UIWidget.h:697]: All right, I'll change it to Node.

**Pull request closed without merging**